src/tangd.c

```diff
@@ -26,9 +26,33 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <getopt.h>
 
 #include <jose/jose.h>
 #include "keys.h"
+#include "socket.h"
+
+static const struct option long_options[] = {
+	{"port", 1, 0, 'p'},
+	{"listen", 0, 0, 'l'},
+	{"version", 0, 0, 'v'},
+	{"help", 0, 0, 'h'},
+	{NULL, 0, 0, 0}
+};
+
+static void print_help(const char *name)
+{
+	fprintf(stderr, "Usage: %s [OPTIONS] <jwkdir>\n", name);
+	fprintf(stderr, "  -p, --port=PORT                 Specify the port to listen (default 9090)\n");
+	fprintf(stderr, "  -l, --listen                    Run as a service and wait for connections\n");
+	fprintf(stderr, "  -v, --version                   Display program version\n");
+	fprintf(stderr, "  -h, --help                      Show this help message\n");
+}
+
+static void print_version(void)
```

Comment:
Nit: please follow the odd style of the file and have it like this:
```
static void
print_version(void)
```

Reply:
done